lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -486,7 +494,20 @@ def refine_tracks_gaussian(
     if overlap_strategy not in ("ignore", "skip", "multiple"):
         raise ValueError("Invalid overlap strategy selected.")
 
-    tracks._validate_single_source("Gaussian refinement")
+    if len(tracks._kymos) > 1:
+        groups, indices = tracks._tracks_by_kymo()
+        groups = [
+            refine_tracks_gaussian(
+                group,
+                window,
+                refine_missing_frames,
+                overlap_strategy,
+                initial_sigma=initial_sigma,
+                fixed_background=fixed_background,
+            )
+            for group in groups
+        ]
+        return KymoTrackGroup([track for _, track in sorted(zip(chain(*indices), chain(*groups)))])
```

Comment:
Nice use of `chain`! It'll also make it work nicely with empty groups.

Just a thought, considering you use this pattern twice, could consider doing something like:
```python
def _apply_to_group(tracks, func, *args, **kwargs):
    groups, indices = tracks._tracks_by_kymo()
    groups = [func(group, *args, **kwargs) for group in groups]
    return KymoTrackGroup([track for _, track in sorted(zip(chain(*indices), chain(*groups)))])
```
Would make it a little easier to test that for some more cases. Then again, we could save that for later, if we end up using the pattern more often.